src/extension/commands/index.ts

```diff
@@ -35,23 +39,58 @@ export function stopBackgroundTask (cell: NotebookCell) {
   return window.showInformationMessage(`${terminal?.name} task terminated!`)
 }
 
-export async function runCLICommand (cell: NotebookCell) {
-  if (!await CliProvider.isCliInstalled()) {
-    return window.showInformationMessage(
-      'Runme CLI is not installed. Do you want to download it?',
-      'Download now'
-    ).then((openBrowser) => openBrowser && env.openExternal(
-      Uri.parse('https://github.com/stateful/runme/releases')
-    ))
+export function runCLICommand(
+  extensionBaseUri: Uri,
+  grpcRunner: boolean,
+  server: RunmeServer,
+  kernel: Kernel
+) {
+ return async function(cell: NotebookCell) {
+    let runmeExecutable: string
+
+    const args = [
+      `--chdir="${path.dirname(cell.document.uri.fsPath)}"`,
+      `--filename="${path.basename(cell.document.uri.fsPath)}"`
+    ]
+
+    const envs: Record<string, string> = { }
+
+    if (grpcRunner) {
+      runmeExecutable = getBinaryPath(extensionBaseUri, os.platform()).fsPath
+
+      envs['RUNME_SERVER_ADDR'] = server.address()
+
+      if (getTLSEnabled()) {
+        envs['RUNME_TLS_DIR'] = getTLSDir()
+      } else {
+        args.push('--insecure')
+      }
+
+      const runnerEnv = kernel.getRunnerEnvironment()
+      if(runnerEnv && runnerEnv instanceof GrpcRunnerEnvironment) {
+        envs['RUNME_SESSION'] = runnerEnv.getSessionId()
+      }
+    } else {
+      if (!await CliProvider.isCliInstalled()) {
```

Comment:
Do you think we still need this @sourishkrout ?

I kept it around for compatibility reasons. But I should be good now to axe it in favor of the bundled binary, no?

Reply:
We should remove it. There's an issue for it: https://github.com/stateful/vscode-runme/issues/288. Let's do it in a separate PR.